personal_profile/tests.py

```diff
@@ -28,16 +33,41 @@ def profile_1(db, user_1):
                                                                content_type='image/jpeg'),
                                 resume=SimpleUploadedFile('test_resume.txt',
                                                           b'these are the contents of the txt file'))
+    profile_1.save()
     return profile_1
 
 
+@pytest.mark.django_db
 class TestProfileModel:
-    # Testing different Personal Profile attributes
-
-    def test_newprofile(self, profile_1, user_1):
-        assert profile_1.company == 'Test Company'
-        assert profile_1.user == user_1
-        assert profile_1.about == 'Test About'
-        assert profile_1.birth_date == datetime.date(1995, 12, 10)
-        assert profile_1.profile_pic is not None
-        assert profile_1.resume is not None
+    # Testing different Personal Profile saving and deletion
+
+    def test_profile_is_saved_correctly(self, profile_1, user_1):
+        test_profile = PersonalProfile.objects.filter(user=user_1).first()
+        assert test_profile.company == profile_1.company
+        assert test_profile.user == profile_1.user
+        assert test_profile.about == profile_1.about
+        assert test_profile.birth_date == profile_1.birth_date
+        assert test_profile.profile_pic == profile_1.profile_pic
+        assert test_profile.resume == profile_1.resume
+
+    def test_profile_is_deleted(self, profile_1, user_1):
+        test_profile = PersonalProfile.objects.filter(user=user_1).first()
+        # profile_1.save()
+        assert test_profile in PersonalProfile.objects.all()
+        test_profile.delete()
+        assert test_profile not in PersonalProfile.objects.all()
+
+    def test_check_placement_of_files(self, profile_1, user_1):
+        # profile_1.save()
+        assert os.path.isfile(GENERAL_IMAGE_PATH)
+        assert os.path.isfile(GENERAL_RESUME_PATH)
+
+
+@pytest.mark.django_db
+class TestProfileUserRelation:
+    # Testing personal profile model and user relation
+
+    def test_profile_is_deleted_when_user_is_deleted(self, profile_1, user_1):
+        profile_1.save()
```

Comment:
You have saved the profile in fixture setup at line 36, so you could remove this line.

Reply:
done